lambdaclass/mirra_backend
# [GH-637] Add endpoints to equip skin to character

## Motivation

We need a way for users to equip items to their characters.
Closes #637 

## Summary of changes

- [Refactor CoM routes and add new items scope in router](https://github.com/lambdaclass/mirra_backend/pull/644/commits/326e5ff0d6c20ccdf758101d00d6d53b387019e6)
- [Add new CoM ItemController and :equip endpoint](https://github.com/lambdaclass/mirra_backend/pull/644/commits/bb7aa5382d45896f2b1e8639c4c337784699d3c2)
- [Add new error cases in fallback_controller](https://github.com/lambdaclass/mirra_backend/pull/644/commits/44535f4079e44abfaa526bd8b1b67e545836fa4f)
- [Move characters config from main config to a separate json file and access via Gateway app](https://github.com/lambdaclass/mirra_backend/pull/644/commits/a002c49fd6576f6583511f6f9fb466e15ad7cfd4)
- [Add CoM characters to seeds](https://github.com/lambdaclass/mirra_backend/pull/644/commits/2f4eb7eeba7949da3d027449e4d0580d122593a7)
- [Add one unit with each available character when creating a new user](https://github.com/lambdaclass/mirra_backend/pull/644/commits/d48e3c4801b5341aeb1234a541c97bb7e505dffc)
- [Add CoM items templates config and use them in seeds](https://github.com/lambdaclass/mirra_backend/pull/644/commits/984485db6b85ec46fc17722cf40d93e38e7f9536)
- [Add buy endpoint for testing purposes](https://github.com/lambdaclass/mirra_backend/pull/644/commits/0a8936af6f20f784d71d342033cef334489a82a4)
- [Add new purchasable? and characters fields to item templates](https://github.com/lambdaclass/mirra_backend/pull/644/commits/d467589278ef2edcac743ce54de17a90471a6339)

## How to test it?

- Create user:
`{:ok, google_user} = GameBackend.Users.find_or_create_google_user_by_email("some_email")`
- Buy item
`curl -X PUT -d item_name="Muflus Gold" http://localhost:4001/curse/users/google_user.user.id/items/buy`
(see `apps/game_backend/priv/items_templates.json` for item names)
- Equip item
`curl -X PUT -d character_name="muflus" -d item_name="Muflus Gold" http://localhost:4001/curse/users/google_user.user.id/items/equip`
(see `apps/game_backend/priv/characters_config.json` for character names)

⚠️ DISCLAIMER ⚠️ 
The `buy item` step is under development still. Will be further improved in another PR (see [issue](https://github.com/lambdaclass/mirra_backend/issues/638))
## Checklist
